### ISSUE TITLE
Are there more intuitive density diagnostics?

### ISSUE THREAD

**Issue by hdrake** [modscripps/mixsea#73]
I've been running some simple and got tripped up by the convention for reporting back the density diagnostic, which is referenced in sequential 1000m bins. As a result, the output density diagnostics `diag['dense']` and `diag['dense_sorted']` are "steppy" and difficult to interpret. This is particularly problematic if the overturns cross the bounds of the reference pressure bins.

As far as I can tell, this does not affect Thorpe scales, the pre-sorting stratification of the overturns, or the dissipation rate estimates because the calculations within each pressure bin use the whole profile (referenced to the center of the bin) and the bin bounds only determine what indices of the diagnostic arrays get written using that reference pressure. One strange outcome that occurs is that if a given overturn is spread across two reference bins, the Thorpse scales & dissipation rates assigned to it will be slightly discontinuous across the bin bounds. Looks weird but isn't necessarily incorrect. Probably the most correct thing to do is to recommend people check how sensitive their results are to dramatically reduce the width of reference pressure bins (so that all outputs smoothly vary with reference pressure).

<img width="399" alt="Screen Shot 2021-10-10 at 6 43 40 PM" src="https://user-images.githubusercontent.com/12971166/136709286-1197f851-421a-456b-b845-1d058201dceb.png">

These figures are produced by the following minimal example (using master branch commit https://github.com/modscripps/mixsea/commit/2c1a27b5b18d8b28f1af17e6d7dbe0f505ef78d8) with a profile that has constant temperature and salinity except for a big overturn in temperature.
```python
import mixsea as mx
import gsw
import numpy as np
import matplotlib.pyplot as plt

z = np.arange(0, 5500., 1.)
p=gsw.p_from_z(-z, 0.)

T = 10.*np.ones_like(z) - z/5000.
T[2000:2500] = 8
T[2500:3000] = 11
S = 35.*np.ones_like(z)

eps, N2, diag = mx.overturn.eps_overturn(z, T, S, 0, 0, return_diagnostics=True, use_ip=False)

plt.figure(1)
plt.plot(gsw.CT_from_t(S, T, p), z, "k-")
plt.plot(diag['CT_sorted'], z, "C0", lw=9, alpha=0.5)
plt.grid(True)
plt.ylim(5500, 0)
plt.ylabel("depth [m]")
plt.xlabel("")

plt.figure(2)
plt.plot(gsw.rho_t_exact(S, T, p), z, "k-", label="in-situ (locally referenced)")
plt.plot(diag['dens'], z, "C0", lw=3, alpha=1, label="referenced in 1000m bins")
plt.plot(diag['dens_sorted'], z, "C1--", lw=2.5, label="sorted and referenced in 1000m bins")
plt.grid(True)
plt.ylim(5500, 0)
plt.ylabel("depth [m]")
plt.ylabel("potential density")
```

**Comment by jessecusack** [comment]
@hdrake, would you be happy if we close this issue after #97 is merged, since it adds the pressure bin width as an optional parameter. It was also partially addressed by #79, which separated out the thorpe_scales function. I _could_ also add an explicit section to the documentation about the steppyness of the density diagnostic.

**Comment by hdrake** [comment]
> @hdrake, would you be happy if we close this issue after #97 is merged, since it adds the pressure bin width as an optional parameter. It was also partially addressed by #79, which separated out the thorpe_scales function. I _could_ also add an explicit section to the documentation about the steppyness of the density diagnostic.

Yes, agreed those changes should do it. I'm not sure we need an explicit warning anymore.
